react-native-reanimated.d.ts

```diff
@@ -52,10 +52,12 @@ declare module 'react-native-reanimated' {
       IDENTITY = 'identity',
     }
 
+    type ExtrapolateParameter = Extrapolate | { extrapolateLeft?: Extrapolate, extrapolateRight?: Extrapolate } | undefined 
```

Comment:
You already handle undefined in `extrapolate?`

Reply:
If you mean `interpolate(translation.y.value, [0, -75], [0, -150], undefined})`

It will work like `Extrapolate.EXTEND`

If you mean `interpolate(translation.y.value, [0, -75], [0, -150], {
        extrapolateLeft: undefined,
})`

It should throw an error

Reply:
I meant that we have this interface:
```ts
    interface InterpolationConfig {
      inputRange: ReadonlyArray<Adaptable<number>>;
      outputRange: ReadonlyArray<Adaptable<number | string>>;
      extrapolate?: ExtrapolateParameter; // <- extrapolate is optional
      extrapolateLeft?: Extrapolate;
      extrapolateRight?: Extrapolate;
    }
```

and this type:
```ts
type ExtrapolateParameter = Extrapolate | { extrapolateLeft?: Extrapolate, extrapolateRight?: Extrapolate } | undefined 
```

Marking extrapolate as optional already adds `| undefined` to the value.

Reply:
Nice catch, fixed ✅

Removed  `| undefined ` from `ExtrapolateParameter `